runtime/plugin/autoread.vim

```diff
@@ -0,0 +1,7 @@
+" autocommands for starting filesystem based file watcher
+
+augroup autoread
+    autocmd!
+    au BufRead,BufWritePost,FileWritePost,FileAppendPost * call v:lua.vim._watch.start(expand("<abuf>"))
+    au BufDelete,BufUnload,BufWritePre,FileWritePre,FileAppendPre * call v:lua.vim._watch.stop(expand("<abuf>"))
+augroup END
```

Comment:
We should implement this in Lua.

Reply:
If `start` and `stop` took in a number rather than a string, I think something like the following could work:
```lua
-- autocommands for starting filesystem based file watcher
local watcher_callback = function(handler)
  return function() handler(tonumber(vim.fn.expand("<abuf>"))) end
end

local watchgroup = vim.api.nvim_create_augroup("watchgroup", { clear = true })
vim.api.nvim_create_autocmd({
  "BufRead", "BufWritePost", "FileWritePost", "FileAppendPost"
}, {
  pattern = "*",
  group = watchgroup,
  callback = watcher_callback(vim._watch.start)
})
vim.api.nvim_create_autocmd({
  "BufDelete", "BufUnload", "BufWritePre", "FileWritePre", "FileAppendPre"
}, {
  pattern = "*",
  group = watchgroup,
  callback = watcher_callback(vim._watch.stop)
})
```